### DISCUSSION TITLE
Adding J1939 PGNs

### DISCUSSION THREAD

**Discussion by g2griz** [canboat/canboat#429] [General]
I am building an RPi-based replacement for the failing OEM multi-function display on our boat, using signalk via OpenPlotter.  The OEM system has two separate CAN busses. One is N2K between the display and an Airmar DST transducer.  The other uses J1939 to send engine parameters to the display.  It looks like I'll need to process 12 parameters contained in 10 different J1939 PGNs.  Based on data collected via candump I have identified the PGNs, data field offsets and their associated scaling.

In studying analyzer code it looks straightforward to add these parameters.  Before making a pull request I wanted to see if there are any issues with adding this to canboat?  (I understand that there is additional work downstream to allow signalk to accept the data.)

Thanks for any comments.

**Comment by keesverruijt** [comment]
Sorry, this is probably stale and very naughty of me not reacting. Yes, J1939 fixes are welcome. By now (June 2026) I suggest you let AI like Claude or Codex help you prepare a PR.

**Comment by keesverruijt** [comment]
I now have notifications on for discussions, if this is still relevant please open an issue, PR or discussion.
